### ISSUE TITLE
[Bug] assert_near does not handle NaN correctly

### ISSUE THREAD

**Issue by molni** [ohsayan/all_asserts#6]
```#[test]
#[should_panic(
    expected = "assertion failed: `0.99999976 is not within epsilon 0.00000011920929 of NaN"
)]
fn test_assert_near_nan() {
    assert_near!(std::f32::NAN, (1.0f32 - f32::EPSILON - f32::EPSILON), f32::EPSILON);
}
```
if wanted, I can provide a bug fix

**Comment by ohsayan** [comment]
Hmm, this needs to be handled. Do you want to work on this?

**Comment by molni** [comment]
Sure, I can do it. I will also create tests with NaNs for all macros (although I think assert_near is the only one affected).

**Comment by ohsayan** [comment]
This was fixed by @molni via #7! Thanks and sorry for the insane delay!
